src/solver/variable/economy/priceCSR.h

```diff
@@ -0,0 +1,300 @@
+    {
+        // Next
+        NextType::initializeFromArea(study, area);
+    }
+
+    void initializeFromLink(Data::Study* study, Data::AreaLink* link)
+    {
+        // Next
+        NextType::initializeFromAreaLink(study, link);
+    }
+
+    void simulationBegin()
+    {
+        // Next
+        NextType::simulationBegin();
+    }
+
+    void simulationEnd()
+    {
+        NextType::simulationEnd();
+    }
+
+    void yearBegin(uint year, unsigned int numSpace)
+    {
+        // Reset the values for the current year
+        pValuesForTheCurrentYear[numSpace].reset();
+        // Next variable
+        NextType::yearBegin(year, numSpace);
+    }
+
+    void yearEndBuild(State& state, unsigned int year)
+    {
+        // Next variable
+        NextType::yearEndBuild(state, year);
+    }
+
+    void yearEnd(uint year, unsigned int numSpace)
+    {
+        // Compute all statistics for the current year (daily,weekly,monthly)
+        pValuesForTheCurrentYear[numSpace].computeAveragesForCurrentYearFromHourlyResults();
+
+        // Next variable
+        NextType::yearEnd(year, numSpace);
+    }
+
+    void computeSummary(std::map<unsigned int, unsigned int>& numSpaceToYear,
+                        unsigned int nbYearsForCurrentSummary)
+    {
+        for (unsigned int numSpace = 0; numSpace < nbYearsForCurrentSummary; ++numSpace)
+        {
+            // Merge all those values with the global results
+            AncestorType::pResults.merge(numSpaceToYear[numSpace] /*year*/,
+                                         pValuesForTheCurrentYear[numSpace]);
+        }
+
+        // Next variable
+        NextType::computeSummary(numSpaceToYear, nbYearsForCurrentSummary);
+    }
+
+    void hourBegin(uint hourInTheYear)
+    {
+        // Next variable
+        NextType::hourBegin(hourInTheYear);
+    }
+
+    void hourForEachArea(State& state, unsigned int numSpace)
+    {
+        // Marginal Price
+        // Note: The marginal price provided by the solver is negative
+        // (naming convention).
+
+        //     MRG. PRICE CSR = “Unsupplied Energy Cost” ( “Unsupplied Energy Cost’ in Antares IF
+
+        //     after adq patch and DTG MRG netting UNSP. ENR CSR  =/=0  ) MRG. PRICE CSR = MRG.
+        //     PRICE_org (== “MRG. PRICE” before any adq patch  IF after adq patch AND DTG MRG
+        //     netting  UNSP. ENR CSR  = 0    )
+        auto isLow = [](double x) { return ::fabs(x) < 1; };
+
+        if (isLow(
+              state.hourlyResults->ValeursHorairesDeDefaillancePositiveCSR[state.hourInTheWeek]))
+        {
+            pValuesForTheCurrentYear[numSpace][state.hourInTheYear]
+              = -state.hourlyResults->CoutsMarginauxHoraires[state.hourInTheWeek];
+        }
+        else
+        {
+            Antares::logs.notice()
+              << state.area->name << " "
+              << state.hourlyResults->ValeursHorairesDeDefaillancePositiveCSR[state.hourInTheWeek];
```

Comment:
This log won't tell anything to user, It should be more clear.
And it should be separated from the following instruction by an empty line.